### PR TITLE
Update Mamba cache to support ArraysCache 

### PR DESCRIPTION
**Pull request by solarpunkin**
Addresses changes introduced in **ml-explore/mlx-lm#842**, which deprecated `MambaCache` for `ArraysCache` in inference loop. Removed `BatchMambaCache` class as `ArraysCache` now supports `merge()` and `extract()`. Older mlx-lm versions still support MambaCache.

Tests

- **Unit Tests:** All prefix cache guard tests passed after updating sampling_params mock to include missing attributes (logprobs, seed, logits_bias, logits_processors).
- **Linter/Type Check:** `ruff` and `mypy` checks passed.

### CODE REVIEW

**Comment by ericcurtin** [comment]
This seems reasonably fine. But we should change to a model where we strictly define what mlx versions we use vllm-metal side. So we are deliberate in what version of mlx we use at any point in time in the commit history.

File: vllm_metal/v1/model_runner.py [hunk 1]
```diff
@@ -284,59 +283,12 @@ def get_stats(self) -> dict:
...
-
-        Returns:
-            MambaCache for the individual request
-        """
-        cache = MambaCache()
-        for i in range(self._cache_size):
-            if self.cache[i] is not None:
-                cache.cache[i] = self.cache[i][idx : idx + 1]
-        return cache
+AnyCache: TypeAlias = KVCache | ArraysCache | Any
```
Comment (LxYuan0420) [inline vllm_metal/v1/model_runner.py:286]: this make the alias effectively type safety lost?

File: vllm_metal/v1/model_runner.py [hunk 2]
```diff
@@ -284,59 +283,12 @@ def get_stats(self) -> dict:
...
 
 
 def _is_mamba_cache(cache: AnyCache) -> bool:
-    """Check if a cache is a Mamba-style cache (ArraysCache or MambaCache)."""
-    return isinstance(cache, (MambaCache, ArraysCache))
+    """Check if a cache is a Mamba-style cache (has .cache attribute)."""
+    return hasattr(cache, "cache") and not isinstance(cache, (KVCache, BatchKVCache))
```
Comment (LxYuan0420) [inline vllm_metal/v1/model_runner.py:291]: Prefer isinstance(cache, ArraysCache) since MambaCache is a subclass of ArraysCache in mlx-lm; this is clearer and avoids accidental matches

```bash
╰─➤  python
Python 3.12.7 (main, Oct 16 2024, 07:12:08) [Clang 18.1.8 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from mlx_lm.models.cache import ArraysCache, MambaCache
>>> print('issubclass:', issubclass(MambaCache, ArraysCache))
issubclass: True
>>> print('isinstance:', isinstance(MambaCache(), ArraysCache))
isinstance: True
```

File: vllm_metal/v1/worker.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 )
 from vllm.logger import init_logger
 from vllm.lora.request import LoRARequest
-from vllm.model_executor import set_random_seed
+from vllm.utils.torch_utils import set_random_seed
```
Comment (LxYuan0420) [inline vllm_metal/v1/worker.py:18]: importing set_random_seed from vllm.utils.torch_utils breaks on vLLM 0.13.0 (our declared minimum); either keep compatibility with the version we declared or bumped the min version to v0.14.0

File: tests/test_platform.py [hunk 4]
```diff
@@ -5,8 +5,8 @@
 
 import pytest
 import torch
-from vllm.attention.backends.registry import AttentionBackendEnum
-from vllm.attention.selector import AttentionSelectorConfig
+from vllm.v1.attention.backends.registry import AttentionBackendEnum
+from vllm.v1.attention.selector import AttentionSelectorConfig
```
Comment (LxYuan0420) [inline tests/test_platform.py:9]: vllm.v1.attention.* imports in vllm_metal/platform.py and tests/test_platform.py do not exist in vLLM 0.13.0; it only exists in 0.14.0?

File: vllm_metal/v1/model_runner.py [hunk 5]
```diff
@@ -432,7 +384,18 @@ def _extract_kv_cache(
...
-    return [cache.extract(idx) for cache in batch_caches]
+    extracted: list[AnyCache] = []
+    for cache in batch_caches:
+        if hasattr(cache, "extract"):
+            extracted.append(cache.extract(idx))
+        elif hasattr(cache, "cache") and not isinstance(cache, BatchKVCache):
+            # Fallback for older ArraysCache/MambaCache versions where .extract is missing
+            new_cache = type(cache)(len(cache.cache))
+            new_cache.cache = [c[idx : idx + 1] for c in cache.cache]
+            extracted.append(new_cache)
```
Comment (LxYuan0420) [inline vllm_metal/v1/model_runner.py:395]: hmm... can you confirm whether cache can include None entries in this path (hybrid/Mamba/ArraysCache)? The fallback list comprehension slices every entry, which would crash on None. If None is possible, we should preserve those entries in the fallback (or add a test showing it cant happen)

**Comment by solarpunkin** [comment]
I see this getting addressed in #110 so it would be better to close this one. 

[PS: I took a lil break from coding due to fatigue]
